src/elmFormatter.ts

```diff
@@ -0,0 +1,59 @@
+type Buffer = {
+  stdout: string;
+  stderr: string;
+};
+
+type Reject = (reason?: any) => void;
+
+const elmFormat = (inputText: string) => {
+  const setUpProcess = (reject: Reject): Process => {
+    const identifier = nova.extension.identifier;
+    const config = nova.workspace.config;
+    const defaultProcess = new Process("elm-format");
+
+    const configPath = config.get(`${identifier}.elmFormatPath`, "string");
+    if (!configPath || configPath.trim() === "") {
+      const message =
+        "Please provide an elm-format executable in Project Settings to enable formatting.";
+      reject(message);
+      return defaultProcess;
+    }
+    const elmFormatPath = nova.path.expanduser(configPath);
+
+    return new Process(elmFormatPath, { args: ["--stdin"], stdio: "pipe" });
+  };
+
+  const writeToStdin = (process: Process, inputText: string) => {
+    const writer = (process.stdin as any).getWriter();
```

Comment:
I could not, for the life of me, get around this. A [WriteableStream is typed as `unknown`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nova-editor-node/index.d.ts#L14-L15) so I had to just convert it to `any` 😞

Reply:
Yeah I've been stuck on these crappy *Stream types exported by Nova. On my [Nix extension](https://github.com/hansjhoffman/nova-nix/blob/master/src/index.ts#L61), I had to format the file on disk rather than in-memory. Less than ideal but I figured something now is better than nothing given that it could easily be updated later.

Reply:
Ah gotcha. When I tried that it moved the cursor to the [end of the document](https://devforum.nova.app/t/keep-cursor-position-after-altering-a-file/1051) which was frustrating. Hopefully that's avoidable.